### PR TITLE
update for updateTable (in options) and customEditor, and FreezeColumn

### PR DESCRIPTION
**Pull request by GBonnaire**
- change args call updateTable (in options) : With this changement, its more easy to run function of jexcel (i.e. executeFormula) for conditional style

- in createCell, add function for custom editor for updateCell after setData()

### CODE REVIEW

**Comment by GBonnaire** [comment]
In jexcel.js to line : 3
Prev : 
`obj.options.updateTable(el, obj.records[j][i], i, j, obj.options.data[j][i], obj.records[j][i].innerText, jexcel.getColumnNameFromId([i, j]));`

By : 
`obj.options.updateTable(obj, obj.records[j][i], i, j, obj.options.data[j][i], obj.records[j][i].innerText, jexcel.getColumnNameFromId([i, j]));`

And

In jexcel.js to line : 923

Add : 
```
            if (obj.options.columns[i].editor && typeof(obj.options.columns[i].editor.updateCell) == 'function' ) {
                value = obj.options.columns[i].editor.updateCell(td, value);
            }
```

Have Good day

File: dist/jexcel.js [hunk 1]
```diff
@@ -0,0 +1,12172 @@
...
+        } else if (obj.options.columns[i].type == 'image') {
+            if (value && value.substr(0, 10) == 'data:image') {
+                var img = document.createElement('img');
+                img.src = value;
+                td.appendChild(img);
+            }
+        } else {
+            if ((''+value).substr(0,1) == '=') {
+                value = obj.executeFormula(value, i, j)
+            }
```
Comment (GBonnaire) [inline dist/jexcel.js:789]: if (obj.options.columns[i].editor && typeof(obj.options.columns[i].editor.updateCell) == 'function' ) {
                value = obj.options.columns[i].editor.updateCell(td, value);
            }

File: dist/jexcel.js [hunk 2]
```diff
@@ -0,0 +1,12172 @@
...
+                    // Handle big text inside a cell
+                    if (obj.records[y][x].innerHTML.length > 200) {
+                        obj.records[y][x].style.whiteSpace = 'pre-wrap'; 
+                    } else {
+                        if (obj.options.wordWrap == false && obj.options.columns[x].wordWrap == false) {
+                            obj.records[y][x].style.whiteSpace = ''; 
+                        }
+                    }
+                }
+
```
Comment (GBonnaire) [inline dist/jexcel.js:3656]: `obj.options.updateTable(obj, obj.records[j][i], i, j, obj.options.data[j][i], obj.records[j][i].innerText, jexcel.getColumnNameFromId([i, j]));`

**Comment by GBonnaire** [comment]
I use lib JS tableHeadFixer.js

And i init this fixer tab lib in onload of JExcel : 
`$(instance).find("table.jexcel").tableHeadFixer({head:true,left : 3, 'z-index':2100});`

In JExcel, i have updated code for apply a correction in scroll. In this function : 
`obj.updateScroll = function`

Add before check top position Y :
```
var correction = 0;
        for(var ite_col in obj.records[0]) {
            if(obj.records[0][ite_col].style.position=="relative") {
                correction = Math.max(correction,obj.records[0][ite_col].getBoundingClientRect().left+obj.records[0][ite_col].getBoundingClientRect().width);
            } else {
                break;
            }
        }
        if(x2<correction) {
            x = x-correction-obj.content.scrollLeft+parseInt(obj.records[0][ite_col].style.left);
        } 
```

**Comment by GBonnaire** [comment]
An other modify :
In function 
`executeFormula = function`

I needed to have the equivalent of COLUMN in excel for customFormula. For that, i have insert var cell with property (row, Col and cellName)

Prev : 
`var evalstring = '';`

Replace by : 
`var evalstring = 'var cell = {col:'+(x+1)+',row:'+(y+1)+', name:"'+jexcel.getColumnNameFromId([x, y])+'"};';`

**Comment by pphod** [comment]
> In jexcel.js to line : 3
> Prev :
> `obj.options.updateTable(el, obj.records[j][i], i, j, obj.options.data[j][i], obj.records[j][i].innerText, jexcel.getColumnNameFromId([i, j]));`
> 
> By :
> `obj.options.updateTable(obj, obj.records[j][i], i, j, obj.options.data[j][i], obj.records[j][i].innerText, jexcel.getColumnNameFromId([i, j]));`
> 
> And
> 
> In jexcel.js to line : 923
> 
> Add :
> 
> ```
>             if (obj.options.columns[i].editor && typeof(obj.options.columns[i].editor.updateCell) == 'function' ) {
>                 value = obj.options.columns[i].editor.updateCell(td, value);
>             }
> ```
> 
> Have Good day

I understand your point, but when dealing with a library that has lot of people using and relying on the updates. So, if I do that I can crash the environment of several people.

In this instance, you can get OBJ from EL, by

el.jexcel

So, basically you can get the same thing with the actual structure and keep compatibility with all people using.

**Comment by pphod** [comment]
Could you kindly submit that based on the most recent version? Thanks.

**Pull request closed without merging**